feat: add NodeJS API to mermaid-cli

aloisklink opened the pull request:
Sorry, one last PR for you to review @MindaugasLaganeckas!

## :bookmark_tabs: Summary

Instead of just using the mmdc CLI, this PR will make it possible to run mermaid-cli from a NodeJS API, e.g.

```js
import { run } from "@mermaid-js/mermaid-cli"
await run("input.mmd", "output.svg", /* {optional options} */)
```

**BREAKING CHANGE**: The CLI (mmdc) entry to this module is now `src/cli.js`. This shouldn't be a big deal, since it was already changed to `src/index.js` in #350.

## :straight_ruler: Design Decisions

This will make testing easier, and allow nodejs packages to call `mmdc` directly, which should simplify a lot of people's code, e.g. see:
  - https://github.com/mermaid-js/mermaid-cli/issues/271#issue-1186458762
  - https://github.com/mermaid-js/mermaid-cli/issues/247
  - https://github.com/mermaid-js/mermaid-cli/issues/154#issuecomment-989828463
  - My library [remark-mermaid-dataurl](https://github.com/aloisklink/remark-mermaid-dataurl)

### Downsides

I've just found when making this PR that you've already commented on this issue: https://github.com/mermaid-js/mermaid-cli/issues/247#issuecomment-1047489580

- New API means more features to maintain
  - Since the CLI run this code internally, we need these features anyway. I've just turned them into functions and made them public.
  - I made sure to state that the "NodeJS API is not covered by semver" in the README.md, we can change it whenever we want, since the CLI is the more important API.
  - You're also welcome to `@-me` (with `@aloisklink`) on any bug reports, and I'll try to make time to fix them.

## :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid-cli/blob/master/CONTRIBUTING.md)
  - The new `cli()` function I added has the wrong indentation to make the PR review easier. If you want, I can either fix the indentation manually before merge. We could also run a `yarn standard --fix` to autofix indentation.
  - Additionally, some of `run()` has to wrong indentation too, to make PR review easier, since I moved some code into a `try... finally...`
- [x] :computer: have added unit/e2e tests (if appropriate)
  - ~Not added since `jest` only has experimental support for ESM currently: https://github.com/facebook/jest/issues/9430, but the CLI tests call this code.~ Added!
- [x] :bookmark: targeted `master` branch

_Edit:_ Rebased on `master` to fix failing tests due to 171f8b7e982d1f523a2aba41015dd95cb5be7546

MindaugasLaganeckas commented:
I am very grateful to you for all your help 😄

MindaugasLaganeckas commented:
Yes, I see the problems, that your work can create for the project in the future. I also see the opportunities 😄 I am willing to accept the PR.
Would it be possible to include a simple automated test in this PR, that demonstrates the new functionality?
Would it be possible (not necessary) with simple tests demonstrate the new functionality in these cases:
#271
#247
#154 ?
Very minimalistic examples would help others get started if they need to 😄

aloisklink commented:
> :smile: I am willing to accept the PR.

:smiley: Awesome :) Again, feel free to `@aloisklink` me on any bug reports about this feature.

> Would it be possible to include a simple automated test in this PR, that demonstrates the new functionality?

Added! Firstly, I had to convert the tests to use ESM not CJS (see https://github.com/mermaid-js/mermaid-cli/pull/354/commits/32e4877405c29847e8002d807941b43bda25d1b4)

It was a good idea to add tests too! I also found two minor bugs (https://github.com/mermaid-js/mermaid-cli/pull/354/commits/5cb74a93c494546414b06fabc31b55affe820108 and https://github.com/mermaid-js/mermaid-cli/pull/354/commits/2b0d3255c615c4468e96208f6ef567ef7046c0dc)

Let me know if you want me to rebase my commits together!

> Would it be possible (not necessary) with simple tests demonstrate the new functionality in these cases

Added! It's basically just the example I added to the README.md

```js
import { run } from "@mermaid-js/mermaid-cli"
await run("input.mmd", "output.svg", / *{optional options} */)
await run("input.md", "output.md", / *{optional options} */)
```

I've only done a basic example of setting the option `{quiet: true}` in the tests, but they are documented using JSDoc.
